pkg/event-manager/handlers_driver.go

```diff
@@ -0,0 +1,256 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package eventmanager
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	apiclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/entity-store"
+	"github.com/vmware/dispatch/pkg/event-manager/gen/models"
+	driverapi "github.com/vmware/dispatch/pkg/event-manager/gen/restapi/operations/drivers"
+	"github.com/vmware/dispatch/pkg/secret-store/gen/client/secret"
+	"github.com/vmware/dispatch/pkg/trace"
+	"github.com/vmware/dispatch/pkg/utils"
+)
+
+var builtInDrivers = map[string]map[string]bool{
+	"vcenter": map[string]bool{
+		"vcenterurl": true,
+	},
+}
+
+// make sure the input includes all required config values
+func validateEventDriver(driver *Driver) error {
+	template, ok := builtInDrivers[driver.Type]
+	if !ok {
+		// custom driver, no validation
+		return nil
+	}
+
+	apiKeyAuth := apiclient.APIKeyAuth("cookie", "header", "cookie")
+	secrets := make(map[string]string)
+	for _, name := range driver.Secrets {
+		resp, err := SecretStoreClient().Secret.GetSecret(&secret.GetSecretParams{
+			SecretName: name,
+			Context:    context.Background(),
+		}, apiKeyAuth)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get secret %s from secret store", name)
+		}
+		for key, value := range resp.Payload.Secrets {
+			secrets[key] = value
+		}
+	}
+
+	for k := range template {
+		if _, ok := driver.Config[k]; ok {
+			continue
+		}
+		if _, ok := secrets[k]; ok {
+			continue
+		}
+		return fmt.Errorf("no configuration field %s in config or secrets", k)
+	}
+
+	return nil
+}
+
+func (h *Handlers) addDriver(params driverapi.AddDriverParams, principal interface{}) middleware.Responder {
+	defer trace.Tracef("name: %s", *params.Body.Name)()
+
+	sp, _ := utils.AddHTTPTracing(params.HTTPRequest, "EventManager.addDriver")
+	defer sp.Finish()
+
+	if err := params.Body.Validate(strfmt.Default); err != nil {
```

Comment:
I wonder how many other places we aren't actually calling "validate"...

Reply:
Yeah :/ during the security review we should look at our approach to validation & consistency as well...